tests/foreman/ui/test_ansible.py

```diff
@@ -400,6 +400,53 @@ def test_positive_set_ansible_role_order_per_hostgroup(self):
         :CaseAutomation: NotAutomated
         """
 
+    @pytest.mark.tier2
+    def test_positive_assign_and_remove_role_to_hostgroup(
+        self, target_sat, module_org, module_location
+    ):
+        """Add and remove functionality for ansible roles in hostgroup
+
+        :id: 5d94a484-92c1-4387-ab92-0649e4c4f907
+
+        :steps:
+            1. Import all roles available by default.
+            2. Create a hostgroup
+            3. Assign hostgroup to the host
+            4. Register a RHEL host to Satellite with given hostgroup.
+            5. Navigate to the new UI for the given hostgroup.
+            6. Add an ansible roles to the hostgroup
+            7. Remove the added ansible roles from the hostgroup
+
+        :expectedresults: The Ansible Role is successfully assign/add/remove from the hostgroup, and no longer shows
+            up on the UI
+        """
+        SELECTED_ROLE = 'RedHatInsights.insights-client'
+        hostgroup = target_sat.api.HostGroup(
+            location=[module_location],
+            organization=[module_org],
+        ).create()
+        host = target_sat.api.Host(
+            organization=module_org, location=module_location, hostgroup=hostgroup
+        ).create()
+        location = host.location.read()
+        organization = host.organization.read()
+        with target_sat.ui_session() as session:
+            session.location.select(location.name)
+            session.organization.select(organization.name)
+            # assign ansible role to host-group
+            target_sat.api.AnsibleRoles().sync(
+                data={'proxy_id': target_sat.nailgun_smart_proxy.id, 'role_names': [SELECTED_ROLE]}
+            )
+            session.host_new.assign_role_to_hostgroup(host.name)
+            # verify ansible role assigned to new UI for given host-group
+            assert (
+                session.hostgroup.read(
+                    host.hostgroup.read().name, widget_names='ansible_roles.assigned_role'
+                )['ansible_roles']['assigned_role'].split('. ')[1]
+            ) == SELECTED_ROLE
+            # remove ansible role to host-group
+            session.host_new.remove_hostgroup_role(host.name)
```

Comment:
Probably a good idea to add an assertion to verify if the role was removed or not.

Reply:
there is no other option provided for hostgroup after removing ansible role. Here no locator is provided to check ansible role is removed

Reply:
You can read the hostgroup again and do the negative assertion here.

Reply:
Unresolved this, as I still have a few more input on this.
You can try assigning more than one role and then just remove one. wdyt?

Reply:
Also, I think there's a bug in stream around assigning ansible roles to a hostgroup. I filed `SAT-24682` for it. Probably that's why prt is failing.

Reply:
@jameerpathan111 Mentioned issue is resolved now, Its ready for verification, and PRT is also passing, resolving this thread.